File: Assn2.py
import sys
import re

input_file = open(sys.argv[1], "r")
for line in input_file:
    line = line.strip()
    match_string = re.match(r"[$][\^]*((([1-9][0-9]{0,2}(,[0-9]{3})*))|0)([.][0-9]{2})?", line);
    if (match_string) and (match_string.group() == line):
        print("Matched: " +line)
    else:
        print("Not Matched: " +line)

File: Assn2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <regex>
using namespace std;

int main(int argc, char* argv[]) {
    ifstream input_file (argv[1]);
    if(input_file.is_open()) {
        string line;
        regex pattern ("(\\$)(\\^)*(([1-9][0-9]{0,2}(,[0-9]{3})*)|0)(\\.[0-9]{2})?");
        while(getline(input_file, line)){
            if (regex_match(line, pattern)) {
                cout << "Matched: " + line << '\n';
            }
            else {
                cout << "Not Matched: " + line << '\n';
            }
        }
        input_file.close();
    } else {
        cout << "Error opening file." << '\n';
    }
    return 0;
}
